src/modules/commander/Commander.cpp

```diff
@@ -2140,18 +2146,31 @@ Commander::run()
 		    failure_detector_updated) {
 
 			if (_failure_detector.isFailure()) {
-				if ((hrt_elapsed_time(&_time_at_takeoff) < 3_s) &&
-				    !_lockdown_triggered) {
+
+				const hrt_abstime time_at_arm = armed.armed_time_ms * 1000;
+
+				if (hrt_elapsed_time(&time_at_arm) < 500_ms) {
+					// 500ms is the PWM spoolup time. Within this timeframe controllers are not affecting actuator_outputs
+
+					if (status.failure_detector_status & vehicle_status_s::FAILURE_ARM_ESC) {
+						arm_disarm(false, true, &mavlink_log_pub, "Failure detector");
+						_status_changed = true;
+						mavlink_log_critical(&mavlink_log_pub, "ESCs did not respond to arm request");
+					}
+
+				} else if (hrt_elapsed_time(&_time_at_takeoff) < (1_s * _param_com_lkdown_tko.get())) {
```

Comment:
Why is this an `else`? It means during the first 500ms you don't check this.

Reply:
I originally made it like this because if you use PWM outputs in the first 500ms the signal is still in the spoolup phase, thus the vehicle has not yet taken off.

This probably does not hold if you use UAVCAN, right?

Reply:
No, and I'd like to avoid these kind of dependencies and implicit assumptions.